Objects/genericaliasobject.c

```diff
@@ -586,6 +587,7 @@ PyTypeObject Py_GenericAliasType = {
     .tp_basicsize = sizeof(gaobject),
     .tp_dealloc = ga_dealloc,
     .tp_repr = ga_repr,
+    .tp_as_number = &_Py_union_as_number,  // allow X | Y of GenericAlias objs
```

Comment:
This is super-weird, sharing of the 'as_number' struct between types. If in the future someone adds support for more numeric operations to unionobject.c, GenericAlias would automatically inherit those, even if inappropriate. I think there should be a separate struct and the function should be exported.

Reply:
Oops, you're right, I noticed typeobject.c also creates its own struct probably for the same reasons too.